doc/devel/contributing.rst

```diff
@@ -543,3 +543,76 @@ example code you can load it into a file handle with::
 
     import matplotlib.cbook as cbook
     fh = cbook.get_sample_data('mydata.dat')
+
+.. _verbose-error-messages:
+
+Error Messages
+--------------
+
+Error messages should err on the side of being verbose, descriptive, and
+specific about what went wrong.  We should be mindful that users are almost
+always using Matplotlib as part of some other goal and may not be (or be
+interested in becoming) experts in the minutiae of our API.  A goal of the
+error message is, when possible, to be informative enough that a typical user
+can understand, debug, and fix their problem based solely on the error without
+the need to consult the online documentation.
+
+One place where it is possible to have extremely informative error messages is
+in the input validation logic.
+
+An example of a **bad** error message ::
+
+  raise ValueError("invalid value")
+
+Which tells the user that we do not like the input, but no hints as to why!
+Because Matplotlib is the base for so many other libraries the user may not
+have directly called Matplotlib or even explicitly generated the invalid value.
+In cases like this it may be possible to get into a debugging and from reading
+the stack trace sort out right local variable to check is to start to sort out
+what went wrong.
```

Comment:
not sure what is meant by "to get into a debugging", and not sure it's made better in context by saying "to get into a debugger", which was my first guess.

Reply:
did mean "debugger".